peachjam/analysis/citations.py

```diff
@@ -27,7 +30,9 @@ def extract_citations(self, document):
     def extract_citations_from_html(self, document):
         html = lxml.html.fromstring(document.content_html)
         for matcher in self.matchers:
-            matcher().markup_html_matches(document.expression_uri(), html)
```

Comment:
So how was this working before? 
Now it's clear tat we are getting the citations from the response and attaching to the doc.html. What magic was this did this line doing? :smile: Was it still calling out to an external service?

Reply:
The other matchers are all in-process and modify `html` directly. Because we're calling an external service now, it can't modify the parsed HTML tree directly. So it can only return a new tree.